api/python/quilt3/api.py

```diff
@@ -162,10 +162,27 @@ def list_packages(registry=None):
     registry_base_path = get_package_registry(fix_url(registry) if registry else None)
 
     named_packages = registry_base_path.rstrip('/') + '/named_packages/'
+    prev_pkg = None
     for path, _ in list_url(named_packages):
         parts = path.split('/')
-        if len(parts) == 3 and parts[2] == 'latest':
-            yield f'{parts[0]}/{parts[1]}'
+        if len(parts) == 3:
+            pkg = f'{parts[0]}/{parts[1]}'
+            # A package can have multiple versions, but we should only return the name once.
+            if pkg != prev_pkg:
+                prev_pkg = pkg
+                yield pkg
+
+
+def list_package_versions(name, registry=None):
+    validate_package_name(name)
```

Comment:
This is a user-facing method, right? We should add doc-string for docs.quiltdata.com

Reply:
I realized that the "versions" this method is returning aren't even useful for anything - need to resolve them to hashes first. Sigh.
And that brings up the question: how would we indicate which hash is "latest"?

Reply:
Maybe by yielding a hash, timestamp pair? 
```
> list(quilt3.list_package_versions("quiltdata/example"))
"quiltdata/example"        00817c1cf85ac050e0c8fb9652a7a6     2019-10-03 12:34:56
"quiltdata/example"        9dd6b60e1ebebef962d3411ede8625     2019-10-01 01:23:45
```